specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/agreement.json

```diff
@@ -67,215 +64,274 @@
+          "x-ms-identifiers": []
+        },
+        "displayName": {
+          "description": "The name of the agreement signed by a customer.",
           "type": "string",
-          "x-ms-enum": {
-            "modelAsString": true,
-            "name": "category"
-          }
+          "readOnly": true
         },
         "effectiveDate": {
-          "description": "The date from which the agreement is effective.",
           "format": "date-time",
-          "readOnly": true,
-          "type": "string"
+          "description": "The date from which the agreement is effective.",
+          "type": "string",
+          "readOnly": true
         },
         "expirationDate": {
-          "description": "The date when the agreement expires.",
           "format": "date-time",
-          "readOnly": true,
-          "type": "string"
+          "description": "The date when the agreement expires.",
+          "type": "string",
+          "readOnly": true
         },
         "participants": {
           "description": "The list of participants that participates in acceptance of an agreement.",
+          "type": "array",
           "items": {
-            "$ref": "#/definitions/Participants"
+            "$ref": "#/definitions/Participant"
           },
-          "type": "array"
+          "readOnly": true,
+          "x-ms-identifiers": []
         },
         "status": {
           "description": "The current status of the agreement.",
+          "enum": [
+            "Unknown",
+            "Draft",
+            "PendingSignature",
+            "Canceled",
+            "Declined",
+            "Active",
+            "Expired",
+            "Terminated",
+            "Deleted",
+            "Submitted"
+          ],
+          "type": "string",
           "readOnly": true,
-          "type": "string"
+          "x-ms-enum": {
+            "name": "AgreementStatus",
+            "modelAsString": true
+          }
+        },
+        "leadBillingAccountName": {
+          "description": "The ID of the lead billing account if this agreement is part of the Customer Affiliate Purchase Terms.",
+          "type": "string",
+          "readOnly": true
         }
       }
     },
     "BillingProfileInfo": {
       "description": "Details about billing profile associated with agreement and available only for specific agreements.",
       "type": "object",
       "properties": {
-        "billingProfileId": {
-          "description": "The unique identifier for the billing profile.",
+        "billingAccountId": {
+          "description": "The fully qualified ID that uniquely identifies a billing account.",
           "type": "string"
         },
         "billingProfileDisplayName": {
-          "description": "The name of the billing profile",
+          "description": "The name of the billing profile for the invoice section or customer.",
+          "type": "string"
+        },
+        "billingProfileFriendlyId": {
```

Comment:
What is this? Isn't this the same as billingProfileId (i.e. the customer-provided ARM id if the customer created the BP)?

Reply:
Looking at the implementation, `Agreement.BillingProfileInfo.BillingProfileId` is the system generated id. `Agreement.BillingProfileInfo.BillingProfileFriendlyId` is the customer-provided ARM id or the system id.